src/args.rs

```diff
@@ -1,8 +1,125 @@
-#[derive(Debug)]
-pub struct Args {}
+use clap::Parser;
+
+#[derive(Parser, Debug, Default)]
+#[command(author, version, about, long_about = None, name = "jperf")]
+pub struct Args {
+	/// Application name
+	#[arg(long)]
+	app: Option<String>,
+
+	/// Cloud name
+	#[arg(long)]
+	cloud: Option<String>,
+
+	/// Unit name
+	#[arg(long)]
+	unit: Option<String>,
+
+	/// Model name
+	#[arg(long)]
+	model: Option<String>,		
+
+	/// Display a large set of metrics
+	#[arg(short, long)]
+	verbose: bool,
+
+	/// Optional choice of a single metric to display
+	#[arg(short, long)]
+	metric: Option<String>,	
+	
+	/// Most recent time interval in milliseconds
+	#[arg(long, default_value_t = 1000)]
+	interval: u32,
+}
 
 impl Args {
-    pub fn parse() -> Self {
-        todo!()
-    }
+	#[allow(dead_code)]
+	fn app(&self) -> &Option<String>{
+		&self.app
+	}
+
+	#[allow(dead_code)]
+	fn unit(&self) -> &Option<String>{
+		&self.unit
+	}
+	
+	#[allow(dead_code)]
+	fn model(&self) -> &Option<String>{
+		&self.model
+	}
+
+	#[allow(dead_code)]
+	fn cloud(&self) -> &Option<String>{
+		&self.cloud
+	}
+
+	#[allow(dead_code)]
+	fn metric(&self) -> &Option<String>{
+		&self.metric
+	}
+
+	#[allow(dead_code)]
+	fn is_verbose(&self) -> &bool {
+		&self.verbose
+	}
+	
+	#[allow(dead_code)]
+	fn interval(&self) -> &u32 {
+		&self.interval
+	}
+
+	#[allow(dead_code)]
+	fn entity_as_string(&self, entity : &str) -> String {		
+		let field : &Option<String> = match entity {
+			"cloud" => self.cloud(),
+			"model" => self.model(),
+			"app" => self.app(),
+			"unit" => self.unit(),
+			&_ => todo!(),
```

Comment:
Let's not panic, please! If this is test code, we should `panic!` with an appropriate error message, otherwise, this should return an `Option<&str>`.

A default `_ => panic` case doesn't feel like good code to keep around as we could add more valid options later, forget to update this and then if we're not careful, we have an avoidable panic on our hands!

Reply:
Yeah, this was something I left behind to take care of as a "todo" item, since I didn't know what to do for that default case.
Following up from the previous comment, the default case accordingly returns `&None`, as Cargo conveniently proposed me to do so :)